workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/executor/request/ExecuteRequestExecutor.java

```diff
@@ -45,6 +52,30 @@ public void execute(ActivityExecutorContext<ExecuteRequest> execution) throws IO
     execution.setOutputVariables(outputs);
   }
 
+  private String encodeQueryParameters(String fullUrl) {
+    // splits by the first occurrence of "?" to get the query params part
+    String[] splitUrl = fullUrl.split("\\?", 2);
+    String encodedUrl = splitUrl[0];
+
+    MultiValueMap<String, String> queryParamsMap =
+        UriComponentsBuilder.fromUriString(fullUrl).build().getQueryParams();
```

Comment:
then cannot we keep using this class, parse the url, clear the query params and set the ones that we encoded
to avoid parsing/building the url manually

Reply:
The query params can be cleaned by setting them to Null, but we can only access them after building the UriComponents, the builder UriComponentsBuilder does not provide getter methods. 
In my last commit, I change the instantiation (fromUriString(fullUrl)) and I rather clone the builder, but I don't really see an interest of doing that.